.github/workflows/documentation.yml

```diff
@@ -0,0 +1,50 @@
+name: Deploy Documentation to GitHub Pages
+
+on:
+  workflow_dispatch: {}
+  push:
+    branches:
+      - master
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
```

Comment:
there is v4 iirc

Reply:
You are correct.

Also, related, removed the "fetch-depth" option since checkout only checks out the latest commit by default. (0 downloads entire history, which we don't need.)